contracts/main/stablecoin-core/StableSwapModuleWrapper.sol

```diff
@@ -74,76 +90,103 @@ contract StableSwapModuleWrapper is PausableUpgradeable, ReentrancyGuardUpgradea
 
-        IToken(token).safeTransferFrom(msg.sender, address(this), _amount6Decimals);
-        IToken(token).approve(stableSwapModule, _amount6Decimals);
-        IStableSwapModule(stableSwapModule).depositToken(token, _amount6Decimals)
-        
-        IToken(stablecoin).safeTransferFrom(msg.sender, address(this), _amount);
-        IToken(stablecoin).approve(stableSwapModule, _amount);
-        IStableSwapModule(stableSwapModule).depositToken(stablecoin, _amount)
-
-        //then call deposit fn for stablecoin
-//deposit tracker is saving only half of total token amount, later when
-        //withdrawl happens
-        // ( depositTracker * 2 ) * ratioOfStablecoin in SSM 
-        // 
-        // ( depositTracker * 2 ) * ratioOfToken in SSM 
-        //need to be sent to withdrawer, of course, for token withdrawl, conversion
-        //of decimals will be done
-         depositTracker[msg.sender] += _amount;
+        depositTracker[msg.sender] += 2 * _amount;
+        totalValueDeposited += 2 * _amount;
 
-    event LogDepositTokens(address indexed _depositor, uint256 _amount);
+        
+        _transferToTheContract(stablecoin, _amount);
+        _transferToTheContract(token, _amount6decimals);
+        _depositToStableSwap(stablecoin, _amount);
+        _depositToStableSwap(token, _amount6decimals);
+        
+        emit LogDepositTokens(msg.sender, _amount);
     }
-
     function withdrawTokens(uint256 _amount) external override nonReentrant whenNotPaused onlyWhitelistedIfNotDecentralized{
-     require(_amount != 0, "depositStablecoin/amount-zero");
-         
+        require(_amount != 0, "depositStablecoin/amount-zero");
+        require(depositTracker[msg.sender] >= _amount, "withdrawTokens/amount-exceeds-deposit");
+        require(totalValueDeposited >= _amount * 2, "withdrawTokens/amount-exceeds-total-deposit");
+        
+        uint256 stablecoinBalanceStableSwap18Decimals = IStableSwapRetriever(stableSwapModule).tokenBalance(stablecoin);
+        uint256 tokenBalanceStableSwap6Decimals = IStableSwapRetriever(stableSwapModule).tokenBalance(token);
+        uint256 tokenBalanceStableSwap18Decimals = _convertDecimals(tokenBalanceStableSwap6Decimals, IToken(token).decimals(), 18);
+       
+        require(stablecoinBalanceStableSwap18Decimals + tokenBalanceStableSwap18Decimals >= _amount, 
+                "withdrawTokens/amount-exceeds-total-deposit");
+
+        uint256 stablecoinAmountToWithdraw = 
+                    _amount * WAD * stablecoinBalanceStableSwap18Decimals
+                    /(stablecoinBalanceStableSwap18Decimals + tokenBalanceStableSwap18Decimals) 
+                    / WAD;
+
+
+        uint256 tokenAmountToWithdraw = 
+                    _amount * WAD * tokenBalanceStableSwap18Decimals
+                    /(stablecoinBalanceStableSwap18Decimals + tokenBalanceStableSwap18Decimals)
+                    /WAD;
+
+        //TODO: below line too risky of due to precision loss?
```

Comment:
yes boss, it can revert because there will be cases when 
stablecoinAmountToWithdraw + tokenAmountToWithdraw
is less than
_amount